### DISCUSSION TITLE
Ergast is shutting down

### DISCUSSION THREAD

**Discussion by SCasanova** [SCasanova/f1dataR#207] [General]
> The Ergast API is now DEPRECATED
> 
> The API will be updated until the end of the 2024 season then shutdown at the end of the year.

This is now posted on the Ergast home page. We should look to find alternatives for that type of info / maybe even capture historical data before the shutdown

**Comment by pbulsink** [comment]
Ergast enables downloads of the whole database in various formats (.csv files, some sort of sql file, etc.). This may be an option. We'll watch and see what FastF1 does too.

**Comment by pbulsink** [comment]
Update: FastF1 is on it, looking at a few different options: https://github.com/theOehrly/Fast-F1/discussions/445

**Comment by pbulsink** [comment]
Should we `deprecate_warn` Ergast functions too, not because they'll all be replaced, but to warn folks that the current package version likely won't work in a year? Once FastF1 gets their Ergast backend up and running we can convert to that API, and remove the depreciation warning.

**Comment by pbulsink** [comment]
A backup option could be to implement basic stats from [motorsportsstats](https://fiaresultsandstatistics.motorsportstats.com/) which apparently also has an api (https://motorsportstats.com/api/result-statistics?sessionSlug=fia-formula-2-championship_2024_jeddah_race-2&sessionFact=LapChart&size=25)... I was linked to it in a discussion on F2 data but I'm sure with some careful research we could identify F1 endpoints.

**Comment by pbulsink** [comment]
Folks loosely affiliated with FastF1 have posted an update:

https://github.com/theOehrly/Fast-F1/discussions/445#discussioncomment-10380182

> Things are moving forward and we've reached alpha state for the Ergast successor. Here's an update, posted by Jess Perry in the Slack Workspace that was mentioned above.
>
>>    Hey everyone,
>>    We’re excited to announce that the Ergast API replacement that we’ve been working on (jolpica-f1) has reached Alpha and is ready for testing. Our focus has been (and will remain for the 2024 season) to reach full feature parity with the existing API before making improvements. We’ve landed on the [Apache 2.0](https://github.com/jolpica/jolpica-f1/blob/main/LICENSE) open source license for this project, which we believe best fits the project’s needs today and into the future.
>>    You can help us out by testing the API and [reporting any bugs you find on GitHub](https://github.com/jolpica/jolpica-f1/issues/new/choose). Please have a look at the “[How to contribute](https://github.com/jolpica/jolpica-f1/blob/main/CONTRIBUTING.md)” notes in the repository as well.
>>    To try it out, simply replace [ergast.com/api/f1/](http://ergast.com/api/f1/) with [api.jolpi.ca/ergast/f1/](http://api.jolpi.ca/ergast/f1/). We currently have an unauthenticated rate limit of 200 requests per hour, which we may adjust based on load, if you have a use case that may need a higher limit please reach out, and we may be able to sort something out.
>>    If you would like to support jolpica-f1 monetarily, we have [set up a Ko-fi](https://ko-fi.com/jolpicaf1). This project is not a for-profit venture, but it will have costs to maintain over time and we’d like to share that burden as much as possible. Our intentions are to be as transparent as possible with this project and that includes its finances.
>>    Special thanks to Philipp Schaefer, Xuanyi Wang and Chris Free for all of their hard work getting us to this point.
>>
>>    We have a few intentional differences from ergast documented in the README & issues, the largest being no longer supporting XML.
>>    If you spot anything not mentioned there, it may be a bug.
>>    We plan to get some better documentation in place soon
>
> Please note, [jolpica-f1](https://github.com/jolpica/jolpica-f1) is a project that is independent of FastF1. I am only a contributor and partially an initiator of the project, and others have done a huge amount of work to make it happen. All issues should be reported in the project's GitHub repository and not here.
>
> Users of FastF1 can manually change the URL for the Ergast API from the default server to the jolpica-f1 server. To do this, your scripts need to contain the following lines of code at the beginning:
>
>```
>import fastf1
>fastf1.ergast.interface.BASE_URL = "https://api.jolpi.ca/ergast/f1"
>```
>
> This will cause FastF1 to use the jolpica-f1 server for internal requests and for all requests that you make explicitly through the Ergast API wrapper.
